src/client/components/Layout.js

```diff
@@ -24,6 +25,38 @@ const Layout = ({ children }) => {
     { name: 'Users', href: '/users', exact: false },
   ];
 
+  //Create a NEW USER & PROFILE after adding a new account in the singInWithPopUp window
+  const createNewUserInFirebase = (newUser) => {
+    const { uid, displayName } = newUser;
+  
+    const userRef = firebase.firestore().collection(USERS).doc(uid);
+    
+    userRef.get()
+    .then((doc) => {
+      if (doc.exists) {
+        console.log("User already exists");
+      } else {
+        const newUser = {
+          displayName,
+          isAdmin: false,
+          uid,
+          about: ''
+        };
+        userRef.set(newUser);
+      }
+    });
+  };
+    
+    const handleClick = () =>{
+      auth.signInWithPopup(googleAuthProvider)
+      .then((data) =>{
+        createNewUserInFirebase(data.user)
+    })
+    .catch((err) =>{
+      console.log(err)
+    })
+  }
+
```

Comment:
Well done! You noticed that there was no way to create a user even though the security rules for this feature were required.